homeassistant/components/here_travel_time/sensor.py

```diff
@@ -0,0 +1,468 @@
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+TRACKABLE_DOMAINS = ["device_tracker", "sensor", "zone", "person"]
+DATA_KEY = "here_travel_time"
+
+NO_ROUTE_ERRORS = ["NGEO_ERROR_GRAPH_DISCONNECTED", "NGEO_ERROR_ROUTE_NO_END_POINT"]
+NO_ROUTE_ERROR_MESSAGE = "HERE could not find a route based on the input"
+
+COORDINATE_SCHEMA = vol.Schema(
+    {
+        vol.Inclusive(CONF_DESTINATION_LATITUDE, "coordinates"): cv.latitude,
+        vol.Inclusive(CONF_DESTINATION_LONGITUDE, "coordinates"): cv.longitude,
+    }
+)
+
+PLATFORM_SCHEMA = vol.All(
+    cv.has_at_least_one_key(CONF_DESTINATION_LATITUDE, CONF_DESTINATION_ENTITY_ID),
+    cv.has_at_least_one_key(CONF_ORIGIN_LATITUDE, CONF_ORIGIN_ENTITY_ID),
+    PLATFORM_SCHEMA.extend(
+        {
+            vol.Required(CONF_APP_ID): cv.string,
+            vol.Required(CONF_APP_CODE): cv.string,
+            vol.Inclusive(
+                CONF_DESTINATION_LATITUDE, "destination_coordinates"
+            ): cv.latitude,
+            vol.Inclusive(
+                CONF_DESTINATION_LONGITUDE, "destination_coordinates"
+            ): cv.longitude,
+            vol.Exclusive(CONF_DESTINATION_LATITUDE, "destination"): cv.latitude,
+            vol.Exclusive(CONF_DESTINATION_ENTITY_ID, "destination"): cv.entity_id,
+            vol.Inclusive(CONF_ORIGIN_LATITUDE, "origin_coordinates"): cv.latitude,
+            vol.Inclusive(CONF_ORIGIN_LONGITUDE, "origin_coordinates"): cv.longitude,
+            vol.Exclusive(CONF_ORIGIN_LATITUDE, "origin"): cv.latitude,
+            vol.Exclusive(CONF_ORIGIN_ENTITY_ID, "origin"): cv.entity_id,
+            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+            vol.Optional(CONF_MODE, default=TRAVEL_MODE_CAR): vol.In(TRAVEL_MODE),
+            vol.Optional(CONF_ROUTE_MODE, default=ROUTE_MODE_FASTEST): vol.In(
+                ROUTE_MODE
+            ),
+            vol.Optional(CONF_TRAFFIC_MODE, default=False): cv.boolean,
+            vol.Optional(CONF_UNIT_SYSTEM): vol.In(UNITS),
+        }
+    ),
+)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: Dict[str, Union[str, bool]],
+    async_add_entities: Callable,
+    discovery_info: None = None,
+) -> None:
+    """Set up the HERE travel time platform."""
+    hass.data.setdefault(DATA_KEY, [])
+
+    app_id = config[CONF_APP_ID]
+    app_code = config[CONF_APP_CODE]
+    if config.get(CONF_ORIGIN_LATITUDE) is not None:
+        origin = ",".join(
+            [str(config[CONF_ORIGIN_LATITUDE]), str(config[CONF_ORIGIN_LONGITUDE])]
+        )
+    else:
+        origin = config[CONF_ORIGIN_ENTITY_ID]
+
+    if config.get(CONF_DESTINATION_LATITUDE) is not None:
+        destination = ",".join(
+            [
+                str(config[CONF_DESTINATION_LATITUDE]),
+                str(config[CONF_DESTINATION_LONGITUDE]),
+            ]
+        )
+    else:
+        destination = config[CONF_DESTINATION_ENTITY_ID]
+
+    travel_mode = config.get(CONF_MODE)
```

Comment:
Please only use `dict.get` if we don't know that the key is in the dict.
```suggestion
    travel_mode = config[CONF_MODE]
```